kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/SessionStateStore.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.statestore
+
+import java.util.concurrent.TimeUnit
+
+import scala.collection.JavaConverters._
+
+import org.apache.kyuubi.client.api.v1.dto.{Batch, BatchRequest}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.ApplicationOperation._
+import org.apache.kyuubi.operation.OperationState
+import org.apache.kyuubi.server.KyuubiRestFrontendService
+import org.apache.kyuubi.server.statestore.api._
+import org.apache.kyuubi.service.AbstractService
+import org.apache.kyuubi.util.ThreadUtils
+
+class SessionStateStore extends AbstractService("SessionStateStore") {
+  private var _stateStore: StateStore = _
+
+  private lazy val restInstance = KyuubiRestFrontendService.getConnectionUrl
```

Comment:
it's a little dangerours for null.  seems better to initialize at start method

Reply:
the frontend services are initialized after backend service. So, we can not init it at start method.